api/v1alpha1/limitador_types.go

```diff
@@ -102,6 +106,31 @@ type LimitadorList struct {
 	Items           []Limitador `json:"items"`
 }
 
+// Storage contains the options for Limitador counters database or in-memory data storage
+type Storage struct {
+	// +optional
+	// +kubebuilder:default="in-memory"
+	Type StorageType `json:"type,omitempty"`
+
+	// +ConfigSecretRef refers to the secret holding the URL for Redis or Inifinispan.
+	// +optional
+	ConfigSecretRef *corev1.ObjectReference `json:"configSecretRef,omitempty"`
+}
+
+// StorageType defines the valid options for storage
+// +kubebuilder:validation:Enum=in-memory;redis;infinispan
+type StorageType string
+
+const (
+	StorageTypeInMemory   StorageType = "in-memory"
+	StorageTypeRedis                  = "redis"
+	StorageTypeInfinispan             = "infinispan"
```

Comment:
We ripped that out of the current artifacts… Today the limitador image being used doesn't have `infinispan` built in. So unsure: 
 - Either we leave that in and expect someone to use an alternate image (which they can do)
 - Or we just don't support infinispan at all 

I'd be in favor of the latter, but willing to be wrong.

Reply:
Should I leave it with `memory` and `redis` only? I could add `redis-cached` too

Reply:
I think I'd do `memory` and `redis` only for this release, but others might disagree… I don't know if someone uses `redis_cached` today tho… @eguzki do you?

Reply:
I do not know :shrug: 

Adding redis-cached stuff shouldn't be hard. It is only about reading more fields from the spec (for which I would create a specific struct). 

```go
// Storage contains the options for Limitador counters database or in-memory data storage
type Storage struct {
	// +optional
	Redis *RedisStorageSpec `json:"redis,omitempty"`
         // +optional
	RedisCached *RedisCachedStorageSpec `json:"redisCached,omitempty"`
}
```
When all of them are empty (null), fallback to in-memory.

I would also add some logic to prevent changing the storage. Storage should only be "install only" and should not reconcile. Moving from some storage type to another means losing all the counters and I doubt anyone would want that. 

Of course, this is something we can do for the second release, not needed now.